src/utils/daos_hdlr.c

```diff
@@ -1568,11 +1600,1358 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+				"\t--src=daos://<pool/cont>\n");
+			D_GOTO(out, rc = EINVAL);
+		}
+	} else {
+		src_type = "path";
+	}
+	if (strncmp(dst, "daos://", 7) == 0) {
+		dst_type = "daos";
+		/* start of dst pool/cont str */
+		dst += 7;
+		if (strncmp(dst, "/", 1) == 0) {
+			fprintf(stderr, "cannot parse daos dst type format, "
+				"please use: \n"
+				"\t--dst=daos://<pool/cont>\n");
+			D_GOTO(out, rc = EINVAL);
+		}
+	} else {
+		dst_type = "path";
+	}
+
+	/* check for src DAOS pool/cont or UNS path */ 
+	if (strcmp(src_type, "daos") == 0) {
+		/* check if this copy is from root of DFS cont */
+		*src_str = strdup("/");
+		*src_dfs_path = strdup(src);
+		rc = fs_copy_parse_uuids(src, &src_saveptr, &(fa->src_p_uuid),
+					 &(fa->src_c_uuid), &src_pool,
+					 &src_cont);
+		if (rc != 0 || (src_pool == NULL || src_cont == NULL)) {
+			fprintf(stderr, "failed to parse source DAOS uuids\n");
+			D_GOTO(out, rc);
+		}
+		int src_len = strlen(*src_dfs_path);
+		int src_pool_cont_len = (strlen(src_pool) +
+					strlen(src_cont) + 1);
+		/* copy from root if equal */
+		if (src_len != src_pool_cont_len) {
+			/* get the container subset path */
+			*src_dfs_path += strlen(src_pool) +
+					strlen(src_cont) + 1;
+			*src_str = strdup(*src_dfs_path);
+		}
+		*src_str_len = strlen(*src_str);
+	} else if (strcmp(src_type, "path") == 0) {
+	        rc = duns_resolve_path(src, src_dattr);
+		if (rc != 0) {
+			/* fs_copy will later check this is a valid src path
+			 * with stat
+			 */
+		  	*src_str = strdup(src);
+		    	src_file_dfs->type = POSIX;
+			rc = 0;
+		} else {
+			*src_str = strdup("/");
+			uuid_copy(fa->src_p_uuid, (*src_dattr).da_puuid);
+			uuid_copy(fa->src_c_uuid, (*src_dattr).da_cuuid);
+		}
+	} else {
+		fprintf(stderr, "cannot parse src format, please use: \n"
+				"--src=<daos>://<pool/cont> | <path>\n"
+				"\ttype is daos, only specified if pool/cont "
+				"used\n");
+		D_GOTO(out, rc = EINVAL);
+	}
+
+	/* check for dst DAOS pool/cont or UNS path */
+	if (strcmp(dst_type, "daos") == 0) {
+		/* check for DAOS pool/cont or UNS path */
+		*dst_str = strdup("/");
+		*dst_dfs_path = strdup(dst);
+		rc = fs_copy_parse_uuids(dst, &dst_saveptr,
+					 &(fa->dst_p_uuid), &(fa->dst_c_uuid),
+					 &dst_pool, &dst_cont);
+		if (rc != 0 || (dst_pool == NULL || dst_cont == NULL)) {
+			fprintf(stderr, "failed to parse destination DAOS "
+				"uuids\n");
+			D_GOTO(out, rc);
+		}
+		int dst_len = strlen(*dst_dfs_path);
+		int dst_pool_cont_len = (strlen(dst_pool) +
+					strlen(dst_cont) + 1);
+		/* copy from root if equal */
+		if (dst_len != dst_pool_cont_len) {
+			/* get the container subset path */
+			*dst_dfs_path += strlen(dst_pool) +
+					strlen(dst_cont) + 1;
+			*dst_str = strdup(*dst_dfs_path);
+		}
+		*dst_str_len = strlen(*dst_str);
+	} else if (strcmp(dst_type, "path") == 0) {
+	        rc = duns_resolve_path(dst, dst_dattr);
+		if (rc != 0) {
+			/* fs_copy will later check this is a valid dst
+			 * path with stat
+ 			*/
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs